lightning/src/util/test_utils.rs

```diff
@@ -224,10 +224,9 @@ impl<'a> chain::Watch<EnforcingSigner> for TestChainMonitor<'a> {
 }
 
 pub struct TestPersister {
-	pub update_ret: Mutex<chain::ChannelMonitorUpdateStatus>,
-	/// If this is set to Some(), after the next return, we'll always return this until update_ret
-	/// is changed:
-	pub next_update_ret: Mutex<Option<chain::ChannelMonitorUpdateStatus>>,
+	/// The queue of update statuses we'll return. If only one is queued, we'll always return it. If
+	/// none are queued, ::Completed will always be returned.
+	pub update_rets: Mutex<VecDeque<chain::ChannelMonitorUpdateStatus>>,
```

Comment:
The semantics around here are a little complicated. I'd recommend doing something like gMock's `WillOnce` and `WillRepeatedly`. But that change may be a bit more involved to be worth it.

Reply:
Yeah, I think I'm inclined to do that.